trieste/acquisition/sampler.py

```diff
@@ -237,7 +237,7 @@ def sample(self, at: TensorType) -> TensorType:
 
         if tf.size(self._eps) == 0:
             self._eps.assign(
-                tf.random.normal([self._sample_size, mean.shape[-1]], dtype=tf.float64)
+                tf.random.normal([self._sample_size, tf.shape(mean)[-1]], dtype=tf.float64)
             )  # [S, L]
```

Comment:
Whats this for? Just for the tensroflow retracing?

Reply:
Yes. Compiling the graph means the variable `mean` only knows the statically determined shape (which may have None in it). You have to use `tf.shape` to get the dynamic shape at each execution.

Reply:
makes sense and sounds familiar